src/input.rs

```diff
@@ -475,23 +499,34 @@ impl<'a, A: ActionContext + 'a> Processor<'a, A> {
         }
     }
 
-    fn scroll_terminal(&mut self, code: u8, modifiers: ModifiersState) {
+    fn scroll_terminal(&mut self, code: u8, modifiers: ModifiersState, scroll_multiplier: u8) {
         debug_assert!(code == 64 || code == 65);
 
-        let faux_scrollback_lines = self.mouse_config.faux_scrollback_lines;
         let mouse_modes = TermMode::MOUSE_REPORT_CLICK | TermMode::MOUSE_DRAG | TermMode::MOUSE_MOTION;
+
+        // Make sure the new and deprecated setting are both allowed
+        let faux_scrollback_lines = self.mouse_config
+            .faux_scrollback_lines
+            .unwrap_or(self.scrolling_config.faux_multiplier as usize);
+
         if self.ctx.terminal_mode().intersects(mouse_modes) {
             self.mouse_report(code, ElementState::Pressed, modifiers);
-        } else if faux_scrollback_lines > 0 {
+        } else if self.ctx.terminal_mode().contains(TermMode::ALT_SCREEN)
+            && faux_scrollback_lines > 0 && !modifiers.shift
+        {
             // Faux scrolling
             let cmd = code + 1; // 64 + 1 = A, 65 + 1 = B
-            let mut content = Vec::with_capacity(faux_scrollback_lines * 3);
+            let mut content = Vec::with_capacity(faux_scrollback_lines as usize * 3);
```

Comment:
I'd go for `faux_scrolling_lines` here as a nitpick instead of `faux_scrollback_lines`. It's an error I've introduced originally but I think it makes a lot more sense to go with scrolling instead of scrollback. This wasn't introduced during rebase but just something I've noticed.